homeassistant/components/weblink.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+"""
+homeassistant.components.weblink
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Adds links to external webpage
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/webpage/
+"""
+
+# The domain of your component. Should be equal to the name of your component
+import logging
+
+from homeassistant.helpers.entity import Entity
+
+DOMAIN = "weblink"
+
+# List of component names (string) your component depends upon
+DEPENDENCIES = []
+
+ATTR_NAME = 'name'
+ATTR_URL = 'url'
+
+DEFAULT_WEBLINK_ICON = 'mdi:open-in-new'
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup(hass, config):
+    """ Setup weblink component. """
+
+    # States are in the format DOMAIN.OBJECT_ID
+
+    links = config.get(DOMAIN)
+
+    for link in links.get('entities'):
+        if ATTR_NAME not in link.keys() or ATTR_URL not in link.keys():
+            _LOGGER.error("You need to set both %s and %s to add a %s",
+                          ATTR_NAME, ATTR_URL, DOMAIN)
+        else:
```

Comment:
Good practice is to use guard clauses and get out of the code when an error occurs. So don't use `else` but use `continue` in case of an error.